packages/live-queries/addon/live-query.js

```diff
@@ -0,0 +1,43 @@
+import Ember from "ember";
+import { getOwner } from '@ember/application';
+import { getProperties } from '@ember/object';
+
+export function liveQuery(...args) {
+  let query;
+  let fn = args.pop();
+  let deps = args;
+  return Ember.computed(...deps, function() {
+    if (query) {
+      // we have already run before, we're recomputing due to a dependent key change
+      query.cleanup();
+    } else {
+      // first time
+      this.on('willDestroyElement', () => query.cleanup());
+    }
+    let inputs = fn(getProperties(this, deps));
+    query = new LiveQuery(inputs, getOwner(this));
```

Comment:
In the readme we mention that we should only use LiveQuery in a component due to the lifecycle operations that a component has. I think we should at the very least emit a warning if this is not within a Component, and probably actually we should throw an error if that is the case.